base/util.jl

```diff
@@ -120,14 +120,14 @@ end
 # print elapsed time, return expression value
 const _mem_units = ["byte", "KB", "MB", "GB", "TB", "PB"]
 const _cnt_units = ["", " k", " M", " G", " T", " P"]
-function prettyprint_getunits(value, numunits, factor)
-    if value == 0 || value == 1
+function prettyprint_getunits(value, num_units, factor)
+    if value == 0
         return (value, 1)
     end
-    unit = ceil(Int, log(value) / log(factor))
-    unit = min(numunits, unit)
-    number = value/factor^(unit-1)
-    return number, unit
+    unit = ceil(Int, log(factor, value + 1))
```

Comment:
In math this works too, but it generates more changes in git diff.

However having read your the other comment, we also have precision issue:

`floor(Int, log(1024, 1024^5-1)) == floor(Int, log(1024, 1024^5+1))`

Maybe I will have to do the loop instead of `log`